modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductMedia.testcase

```diff
@@ -320,6 +320,226 @@ definition {
 		}
 	}
 
+	@description = "COMMERCE-12773 | Verify that it is possible to use the toggle Shopping Experience Image Gallery to decide which image shows on the storefront product details"
+	@priority = 4
+	test CanUseTheShoppingExperienceImageGalleryToggle {
+		property custom.properties = "jsonws.web.service.paths.excludes=";
+		property portal.acceptance = "false";
+
+		task ("Given a Minium Site") {
+			CommerceAccelerators.initializeNewSiteViaAccelerator(siteName = "Minium");
+		}
+
+		task ("And an Account with the buyer") {
+			CommerceEntry.addAccountEntryUser(
+				accountName = "Commerce Account",
+				accountType = "Business",
+				agreeToTermsAndAnswerReminderQuery = "true",
+				createNewAccount = "true",
+				requireReset = "false",
+				userEmailAddress = "buyer@liferay.com",
+				userFirstName = "Buyer",
+				userLastName = "User",
+				userRole = "Buyer",
+				userScreenName = "buyeruser",
+				userSiteMembership = "Minium");
+		}
+
+		task ("And create a new product with an option linked to it") {
+			CommerceJSONProductsAPI._addCommerceProduct(
+				catalogName = "Minium",
+				productName = "Simple_Product",
+				productType = "simple",
+				sku = "TESTSKU");
+
+			CommerceJSONOptionsAPI._assignOptionToCommerceProduct(
+				fieldType = "select",
+				optionKey = "package-quantity",
+				optionName = "Package Quantity",
+				productName = "Simple_Product");
+		}
+
+		task ("And open the product admin details") {
+			CommerceProducts.openProductsAdmin();
+
+			CommerceProducts.searchAndGoToProductEntry(entryProductName = "Simple_Product");
+		}
+
+		task ("And all SKUs are generated") {
+			CommerceEntry.gotoMenuTab(menuTab = "SKUs");
+
+			CommerceSKUs.generateAllSkuCombinations();
+		}
+
+		task ("And disable the toggle Shopping Experience Image Gallery for an image") {
+			CommerceEntry.gotoMenuTab(menuTab = "Media");
+
+			CommerceProductImages.addProductImages(
+				imageTitles = "Black",
+				optionValues = 6,
+				shoppingExperienceImageGallery = "false");
+
+			CommerceProductImages.addProductImages(
+				imageTitles = "White,Blue",
+				optionValues = "12,24",
+				shoppingExperienceImageGallery = "true");
+		}
```

Comment:
This part I think can be improved since the Acceptance Criteria 'By Default set to true', I didn't see it as a part of the assertion in the test.